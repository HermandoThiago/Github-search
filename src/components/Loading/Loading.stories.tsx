import { Meta, Story } from '@storybook/react';
import Loading from './Loading';

export default {
    title: 'Components/Loading',
    component: Loading
} as Meta;

export const Default: Story = () => (
    <Loading />
);