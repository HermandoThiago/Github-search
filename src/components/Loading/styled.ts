import styled from "styled-components";

export const StyledLoader = styled.div`
    width: 100%;
    height: 350px;
    display: flex;
    justify-content: center;
    align-items: center;
`;