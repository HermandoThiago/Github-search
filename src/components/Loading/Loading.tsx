import { StyledLoader } from "./styled";

import { FallingLines } from "react-loader-spinner";

export function Loading() {
  return (
    <StyledLoader>
      <FallingLines color="#fff" width="100" visible={true} />
    </StyledLoader>
  );
}
