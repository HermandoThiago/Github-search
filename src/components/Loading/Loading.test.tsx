import renderer from 'react-test-renderer';
import Loading from './Loading';

describe('Tests Loading', () => {
    it('should match snapshot', () => {
        const component = renderer.create(
            <Loading />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});