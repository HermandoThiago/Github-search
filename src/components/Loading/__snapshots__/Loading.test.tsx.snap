// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Loading should match snapshot 1`] = `
<div
  className="sc-bczRLJ enopnZ"
>
  <svg
    aria-busy={true}
    data-testid="falling-lines"
    height="100"
    role="status"
    viewBox="0 0 100 100"
    width="100"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      data-testid="falling-lines-rect-1"
      fill="#fff"
      height="50"
      rx="4"
      ry="4"
      width="10"
      y="25"
    >
      <animate
        attributeName="x"
        dur="1.2s"
        repeatCount="indefinite"
        values="10;100"
      />
      <animateTransform
        attributeName="transform"
        dur="1.2s"
        from="0 10 70"
        repeatCount="indefinite"
        to="-60 100 70"
        type="rotate"
      />
      <animate
        attributeName="opacity"
        dur="1.2s"
        repeatCount="indefinite"
        values="0;1;0"
      />
    </rect>
    <rect
      fill="#fff"
      height="50"
      rx="4"
      ry="4"
      width="10"
      y="25"
    >
      <animate
        attributeName="x"
        begin="0.4s"
        dur="1.2s"
        repeatCount="indefinite"
        values="10;100"
      />
      <animateTransform
        attributeName="transform"
        begin="0.4s"
        dur="1.2s"
        from="0 10 70"
        repeatCount="indefinite"
        to="-60 100 70"
        type="rotate"
      />
      <animate
        attributeName="opacity"
        begin="0.4s"
        dur="1.2s"
        repeatCount="indefinite"
        values="0;1;0"
      />
    </rect>
    <rect
      data-testid="falling-lines-rect-2"
      fill="#fff"
      height="50"
      rx="4"
      ry="4"
      width="10"
      y="25"
    >
      <animate
        attributeName="x"
        begin="0.8s"
        dur="1.2s"
        repeatCount="indefinite"
        values="10;100"
      />
      <animateTransform
        attributeName="transform"
        begin="0.8s"
        dur="1.2s"
        from="0 10 70"
        repeatCount="indefinite"
        to="-60 100 70"
        type="rotate"
      />
      <animate
        attributeName="opacity"
        begin="0.8s"
        dur="1.2s"
        repeatCount="indefinite"
        values="0;1;0"
      />
    </rect>
  </svg>
</div>
`;
